tests/writer_test.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+"""Tests for the artifact definitions readers."""
+
+import io
+import os
+import unittest
+import json
+import tempfile
+
+from artifacts import reader
+from artifacts import writer
+
+
+class YamlArtifactsWriterTest(unittest.TestCase):
+  """Class to test the YAML artifacts writer"""
+
+  def testYamlWriter(self):
+    """Tests the YamlArtifactsWriter FormatArtifacts method for loss during conversion."""
+    artifact_reader = reader.YamlArtifactsReader()
+    artifact_writer = writer.YamlArtifactsWriter()
+    test_file = os.path.join('test_data', 'definitions.yaml')
+
+    artifact_definitions = list(artifact_reader.ReadFile(test_file))
+    with tempfile.NamedTemporaryFile() as artifact_file:
+      artifact_writer.WriteArtifactsFile(artifact_definitions,
+                                         artifact_file.name)
+      converted_artifact_definitions = list(artifact_reader.ReadFile(
+          artifact_file.name))
+
+    self.assertListEqual(
+        [artifact.AsDict() for artifact in artifact_definitions],
+        [artifact.AsDict() for artifact in converted_artifact_definitions])
+
+
+class JsonArtifactsWriterTest(unittest.TestCase):
+  """Class to test the JSON artifacts writer"""
+
+  def testJsonWriter(self):
+    """Tests the JsonArtifactsWriter FormatArtifacts method for loss during conversion."""
+    artifact_reader = reader.YamlArtifactsReader()
+    artifact_writer = writer.JsonArtifactsWriter()
+    test_file = os.path.join('test_data', 'definitions.yaml')
+
+    artifact_definitions = list(artifact_reader.ReadFile(test_file))
```

Comment:
Half-implementing JSON support is not ideal. I assume you do this because your GUI code will consume it? I think if you write a JSON writer, others users of the library will expect to find a reader.